src/tzdata/compile.jl

```diff
@@ -508,90 +539,101 @@ function resolve!(zone_name::AbstractString, zoneset::ZoneDict, ruleset::RuleDic
-            end
+        if kind == "Rule"
+            rule = parse(Rule, line)
+            rules[name] = push!(get(rules, name, Rule[]), rule)
+        elseif kind == "Zone"
+            zone = parse(Zone, line)
+            zones[name] = push!(get(zones, name, Zone[]), zone)
+        elseif kind == "Link"
+            target = name
+            link_name = line
+            links[link_name] = target
+        else
+            @warn "Unhandled line found with type: $kind"
         end
     end
 
-    # Turn links into zones.
-    # Note: it would be more computationally efficient to pass the links around
-    # and only resolve a zone once.
-    for (dest, source) in links
-        zones[dest] = zones[source]
+    return tz_source
+end
+
+function load!(tz_source::TZSource, file::AbstractString)
+    open(file, "r") do io
+        load!(tz_source, io)
     end
+end
 
-    return zones, rules
+function compile(name::AbstractString, tz_source::TZSource; kwargs...)
+    ordered = OrderedRuleDict()
+    if name in keys(tz_source.links)
+        tz = compile!(tz_source.links[name], tz_source, ordered; kwargs...)
+        rename(tz, name)
+    else
+        compile!(name, tz_source, ordered; kwargs...)
+    end
 end
 
-function load(tz_source_dir::AbstractString=TZ_SOURCE_DIR; max_year::Integer=MAX_YEAR)
-    time_zones = Dict{AbstractString,TimeZone}()
-    for filename in readdir(tz_source_dir)
-        zones, rules = tzparse(joinpath(tz_source_dir, filename))
-        merge!(time_zones, resolve(zones, rules; max_year=max_year))
+function compile(tz_source::TZSource; kwargs...)
+    time_zones = Vector{TimeZone}()
+    ordered = OrderedRuleDict()
+    lookup = Dict{String, TimeZone}()
+    for zone_name in keys(tz_source.zones)
+        tz = compile!(zone_name, tz_source, ordered; kwargs...)
+        push!(time_zones, tz)
+        lookup[zone_name] = tz
     end
+
+    # Convert links into time zones.
+    for (link_name, target) in tz_source.links
+        if !haskey(lookup, link_name) && haskey(lookup, target)
+            push!(time_zones, rename(lookup[target], link_name))
+        elseif !haskey(lookup, target)
+            error("Unable to resolve link \"$link_name\" referencing \"$target\"")
+        end
+    end
+
     return time_zones
 end
 
+
+
 function compile(tz_source_dir::AbstractString=TZ_SOURCE_DIR, dest_dir::AbstractString=COMPILED_DIR; max_year::Integer=MAX_YEAR)
-    time_zones = load(tz_source_dir; max_year=max_year)
+    tz_source_paths = joinpath.(tz_source_dir, readdir(tz_source_dir))
+    time_zones = compile(TZSource(tz_source_paths); max_year=max_year)
 
     isdir(dest_dir) || error("Destination directory doesn't exist")
     empty!(TIME_ZONES)
 
-    for (name, tz) in time_zones
-        parts = split(name, "/")
-        tz_dir, tz_file = joinpath(dest_dir, parts[1:end-1]...), parts[end]
+    for tz in time_zones
+        parts = split(TimeZones.name(tz), '/')
+        tz_path = joinpath(dest_dir, parts...)
```

Comment:
Are you splitting on slashes and splatting into `joinpath` so that Windows paths are normalized from `/` separation to `\`? If so, I believe you can just make this
```julia
tz_path = normpath(dest_dir, TimeZones.name(tz))
```

Reply:
I am converting `/` into the file separator. I do like the idea of switching this to `normpath` but I'm hesitant as the documentation for the function does not mention this behaviour.

Reply:
Oh you're right, it doesn't. That's odd. Do whatever feels right.